### PR TITLE
More various fixes

### PR DESCRIPTION
**Pull request by zupo**
A few more fixes I found when working on a new project.

Again tested in our CI on Linux and macOS.

### CODE REVIEW

File: overrides/default.nix [hunk 1]
```diff
@@ -441,7 +441,7 @@ lib.composeManyExtensions [
         old: {
           postPatch = ''
             substituteInPlace setup.py \
-              --replace 'setup_requires="setupmeta"' 'setup_requires=[]'
+              --replace 'setup_requires="setupmeta"' 'setup_requires=[]' \
```
Comment (zupo) [inline overrides/default.nix:444]: 🤦

File: overrides/default.nix [hunk 2]
```diff
@@ -466,6 +466,22 @@ lib.composeManyExtensions [
...
+        }
+      );
+      flake8-print = super.flake8-print.overridePythonAttrs (
+        old: {
+          postInstall = ''
+            rm -f $out/lib/python3.9/site-packages/pyproject.toml
+          '';
+        }
+      );
+
```
Comment (zupo) [inline overrides/default.nix:484]: How can this be done better? I opted to go with python3.9 as that's what's in nixpkgs-21.11, but it's far from perfect.

Feel free to throw out of the PR.

Comment (zupo) [reply overrides/default.nix:484]: This is the error I see without this override:

```
error: collision between `/nix/store/cwm9fdp3w2zk7p50xab5q3dmklrbv62m-python3.9-flake8-print-4.0.0/lib/python3.9/site-packages/pyproject.toml' and `/nix/store/gcbxi1fqzkbn49qzr7c3lcbp44slpnqd-python3.9-flake8-debugger-4.0.0/lib/python3.9/site-packages/pyproject.toml'
```

Comment (adisbladis) [reply overrides/default.nix:484]: I feel like we should add a hook that cleans up any files dumped directly in the site-packages directory.

Comment (zupo) [reply overrides/default.nix:484]: Yea that would be very handy!

Comment (adisbladis) [reply overrides/default.nix:484]: See #551

Comment (zupo) [reply overrides/default.nix:484]: Fantastic, I'll rebase this PR once #551 is merged.

**Comment by zupo** [comment]
@adisbladis: rebased off of latest master, ready for re-review.

**Pull request merged**